### DISCUSSION TITLE
Anti-aliasing or multisampling in rendering options?

### DISCUSSION THREAD

**Discussion by dcogburn98** [StereoKit/StereoKit#457] [Q&A] [answered]
I'm very new to Stereokit and VR, and 3D development in general. I use VR daily for development of standard desktop apps because I can have a couple screens going at once in it. I never get sick or anything using it, but when I made a floor plane and added a texture to it in Stereokit, I noticed the floor made awful flickering artifacts when you got further away. This flickering gave me motion sickness and I could only use it for a few seconds before I had to take the headset off. I tried several different textures, thinking maybe it's just the ones I've picked so far, but they all seem to do it (albeit some worse than others).

Really I just want to know if there's any built in settings for anti-aliasing. Some search around the website and on google only comes up with the release that says anti-aliasing was implemented in flatscreen mode. I would greatly appreciate any advice!

**Comment by maluoi** [comment, marked as answer]
In this particular case, you're likely looking more for texture sampling options! Big planes with a texture on them are a bit of a worst case scenario for aliasing caused by downsampling, and this is typically solved with mip-mapping.

Right now, StereoKit will only generate mip-maps on textures that have dimensions that are powers of two! I suspect your texture may be missing this detail, causing StereoKit to work entirely without mip-maps. Make sure your texture dimensions are a power of two, and see if that helps! So like, 512x512, or 128x256, or 1024x1024 or something.

If you'd like to learn more about mip-mapping in this context, [Ben Golus has a really fantastic article](https://bgolus.medium.com/sharper-mipmapping-using-shader-based-supersampling-ed7aadb47bec) that gets really into the details!

If that's not it, let me know. StereoKit doesn't do AA in headsets itself, but many XR runtimes _do_ already do supersampling instead due to how they compensate for headset optics, so it often isn't necessary or helpful.

**Reply by dcogburn98** [reply]
Amazing! Just like flipping a switch, I can finally stand to look around in the world for more than just a few seconds! Thanks for the quick reply and easy solution. I've heard of mip-mapping before, but never really understood what exactly it did. I'll be reading Ben Golus' article on it now!
